tiny_dnn/core/framework/tensor.h

```diff
@@ -140,52 +140,32 @@ class Tensor {
         storage_ptr_ = std::make_shared<TensorStorageType>(shape);
     }
 
-    /**
-     * Constructor that accepts a pointer to existing TensorStorage, together
-     * with shape and offset.
-     * @param storage pointer to TensorStorage
-     * @param offset offset from first element of storage
-     * @param shape shape of the Tensor
-     * @return
-     */
-    explicit Tensor(const TensorStoragePointer storage,
-                    const size_t offset,
-                    const std::array<size_t, kDimensions> &shape) {
-        offset_ = offset;
-        shape_ = shape;
-        size_ = product(shape);
-        storage_ptr_ = storage;
-    }
-
     ~Tensor() = default;
 
-    Tensor(const Tensor&other) { //TODO(Randl):deep copy
-        //TODO(Randl)
-        /*other.fromDevice();
+    //TODO(Randl): implement copy and move constructors
+#if 0
+    //TODO(Randl):deep copy
+    Tensor(const Tensor&other) {
+        other.fromDevice();
         shape_ = other.shape_;
         storage_pointer_ = other.storage_pointer_;
         data_is_on_host_ = true;
-        data_dirty_ = true; */
+        data_dirty_ = true;
         //device_data_ is intentionally left uninitialized.
     }
-/*
- *
- * TODO(Randl): Move constructors for Tensor and TensorStorage
-    Tensor &operator = (const Tensor& other) {
-    }
+    //TODO(Randl): Move constructors for Tensor and TensorStorage
+    Tensor &operator = (const Tensor& other) {}
```

Comment:
You should remove or comment out this constructor. VS2013 will call this and make an uninitialized instance.